agent/taskresource/firelens/firelensconfig_unix.go

```diff
@@ -155,6 +170,25 @@ func (firelens *FirelensResource) generateConfig() (generator.FluentConfig, erro
 	return config, nil
 }
 
+// addHealthcheckSections adds a health check input section and a health check output section to the config.
+func (firelens *FirelensResource) addHealthcheckSections(config generator.FluentConfig) {
+	// Health check supported is only added for fluentbit.
+	if firelens.firelensConfigType != FirelensConfigTypeFluentbit {
```

Comment:
I don't think this restriction makes sense. We're blocking customers from using this feature even if they bring their own container.

Reply:
Customer's can add a health check for Fluentd, it just has to be done manually.

We add the health check set up because it will work with our `amazon/aws-for-fluent-bit` image. 

For Fluentd, customers would need to modify the public fluentd docker images to support a health check. We will probably provide suggestions on how to do this. But since we can not provide a nice end-to-end experience for fluentd, we do not add a health check configuration by default.

Reply:
>  We're blocking customers from using this feature even if they bring their own container.

if customers bring their own firelens container, they can have an external config file in the image that has the health check input section (as supported in https://github.com/aws/amazon-ecs-agent/pull/2179), and then they can use a health check command base on that. so i guess it's not blocking them, just that we are not doing things for them in that case.

Reply:
I missed that they could add their own configuration too and bypass ours entirely.